quinn-proto/src/connection/mod.rs

```diff
@@ -887,18 +886,16 @@ where
     ///
     /// The return value if `Ok` contains the bytes and their offset in the stream.
     pub fn read_unordered(&mut self, id: StreamId) -> Result<Option<(Bytes, u64)>, ReadError> {
-        Ok(self.streams.read_unordered(id)?.map(|result| {
-            self.add_read_credits(id, result.max_stream_data, result.max_data);
-            result.result
-        }))
+        let result = self.streams.read_unordered(id);
+        self.post_read(id, result.as_ref().ok().map(|x| x.as_ref()).flatten());
```

Comment:
Could we move more of the repetitive stuff into `post_read()` here?

Reply:
Good call; done.